kernel/aster-nix/src/fs/ext2/impl_for_vfs/inode.rs

```diff
@@ -131,9 +131,22 @@ impl Inode for Ext2Inode {
         Ok(self.create(name, type_, mode.into())?)
     }
 
-    fn mknod(&self, name: &str, mode: InodeMode, dev: Arc<dyn Device>) -> Result<Arc<dyn Inode>> {
-        let inode = self.create(name, InodeType::from(dev.type_()).into(), mode.into())?;
-        inode.set_device_id(dev.id().into()).unwrap();
+    fn mknod(
+        &self,
+        name: &str,
+        mode: InodeMode,
+        inode_type: InodeType,
+        dev: Option<Arc<dyn Device>>,
+    ) -> Result<Arc<dyn Inode>> {
```

Comment:
In this API, there is a potential invariant that "if inode_type == InodeType::NamedPipe, then dev must be None", right? This invariant leaves the caller to maintain, which is not recommended in Rust. One consequence is in ext2's mknod implementation, if it's a `NamedPipe`, then `dev` will never be used.
So we need to rethink the ability of `mknod()` and bound these invariants to Rust types without any intermediate state leaked, for example: introduce a `pub enum MknodType { PipeNode(InodeType), DeviceNode(InodeType, Arc<dyn Device>) }` or something better.